R/data_interface.R

```diff
@@ -239,10 +239,12 @@ spark_read_parquet <- function(sc,
                                columns = NULL,
                                schema = NULL,
                                ...) {
-  c(name, path) %<-% spark_read_compat_param(sc, name, path)
+  params <- spark_read_compat_param(sc, name, path)
+  name <- params[1L]
+  path <- params[-1L]
   if (overwrite) spark_remove_table_if_exists(sc, name)
 
-  df <- spark_data_read_generic(sc, list(spark_normalize_path(path)), "parquet", options, columns, schema)
+  df <- spark_data_read_generic(sc, lapply(path, spark_normalize_path), "parquet", options, columns, schema)
```

Comment:
It would also make sense to vectorize `spark_normalize_path` to support vectors, I would change this: https://github.com/rstudio/sparklyr/blob/787fe0b53dd402490fde0f32ac48f4b4d1af1861/R/utils.R#L151-L159 to:

```r
spark_normalize_single_path <- function(path) {
  # don't normalize paths that are urls
  if (grepl("[a-zA-Z]+://", path)) {
    path
  }
  else {
    normalizePath(path, mustWork = FALSE)
  }
}

spark_normalize_path <- function(paths) {
  unname(sapply(paths, spark_normalize_single_path))
}
```

Reply:
Thanks for the comment, I give this a shot in https://github.com/rstudio/sparklyr/pull/2118/commits/d075bd48313511bfe39fa2f210e688e9837fb30a, looks fine to me, at least all tests pass.